js/piwik.js

```diff
@@ -3902,9 +3902,10 @@ if (typeof window.Matomo !== 'object') {
                     return '';
                 }
 
-                if (!configFileTracking && windowAlias.location.protocol === 'file:') {
-                    return '';
-                }
+              const fileRegex = new RegExp('/^file:\/\/*/i');
+              if (!configFileTracking && (windowAlias.location.protocol === 'file:' || fileRegex.test(currentUrl))) {
+                return '';
+              }
```

Comment:
Might not be that much of an issue, but please try to keep the code alignments in place. It will get harder and harder to read such a file if the indents differ here and there.